clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -17146,21 +17146,14 @@ Decl *Sema::BuildStaticAssertDeclaration(SourceLocation StaticAssertLoc,
                        FoldKind).isInvalid())
       Failed = true;
 
-    // If the static_assert passes, only verify that
-    // the message is grammatically valid without evaluating it.
-    if (!Failed && AssertMessage && Cond.getBoolValue()) {
-      std::string Str;
-      EvaluateStaticAssertMessageAsString(AssertMessage, Str, Context,
-                                          /*ErrorOnInvalidMessage=*/false);
-    }
-
```

Comment:
This bit is added in 47ccfd7a89e2a (https://reviews.llvm.org/D154290) on 2023-07-19, after the commit that is being double cherry-picked added the chunk of code below.